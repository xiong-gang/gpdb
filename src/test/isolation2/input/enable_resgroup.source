-- start_ignore
! rmdir @cgroup_mnt_point@/cpu/gpdb;
! rmdir @cgroup_mnt_point@/cpuacct/gpdb;
! mkdir @cgroup_mnt_point@/cpu/gpdb;
! mkdir @cgroup_mnt_point@/cpuacct/gpdb;
-- end_ignore

-- enable resource group and restart cluster.
-- start_ignore
! gpconfig -c gp_resource_manager -v group;
! gpconfig -c gp_resource_group_cpu_limit -v 0.5;
! gpstop -rai;
-- end_ignore

SHOW gp_resource_manager;

-- resource queue statistics should not crash
SELECT * FROM pg_resqueue_status;
SELECT * FROM gp_toolkit.gp_resqueue_status;
SELECT * FROM gp_toolkit.gp_resq_priority_backend;

-- by default admin_group has concurrency set to -1 which leads to
-- very small memory quota for each resgroup slot, correct it.
ALTER RESOURCE GROUP admin_group SET concurrency 2;
