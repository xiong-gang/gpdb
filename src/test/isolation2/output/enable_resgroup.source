-- start_ignore
! rmdir @cgroup_mnt_point@/cpu/gpdb;

! rmdir @cgroup_mnt_point@/cpuacct/gpdb;

! mkdir @cgroup_mnt_point@/cpu/gpdb;

! mkdir @cgroup_mnt_point@/cpuacct/gpdb;

! chmod 755 @cgroup_mnt_point@/cpu/gpdb;

! chmod 755 @cgroup_mnt_point@/cpuacct/gpdb;

-- end_ignore

-- enable resource group and restart cluster.
-- start_ignore
! gpconfig -c gp_resource_manager -v group;
20170502:01:28:12:000367 gpconfig:sdw6:gpadmin-[WARNING]:-Managing queries with resource groups is an experimental feature. A work-in-progress version is enabled.
20170502:01:28:13:000367 gpconfig:sdw6:gpadmin-[INFO]:-completed successfully

! gpconfig -c gp_resource_group_cpu_limit -v 0.9;
20170731:09:42:33:021079 gpconfig:sdw8:nyu-[INFO]:-completed successfully

! gpconfig -c max_connections -v 125 -m 25;
20170731:09:42:34:021163 gpconfig:sdw8:nyu-[INFO]:-completed successfully

! gpstop -rai;
-- end_ignore

SHOW gp_resource_manager;
gp_resource_manager
-------------------
group              
(1 row)

-- resource queue statistics should not crash
SELECT * FROM pg_resqueue_status;
rsqname|rsqcountlimit|rsqcountvalue|rsqcostlimit|rsqcostvalue|rsqwaiters|rsqholders
-------+-------------+-------------+------------+------------+----------+----------
(0 rows)
SELECT * FROM gp_toolkit.gp_resqueue_status;
queueid|rsqname|rsqcountlimit|rsqcountvalue|rsqcostlimit|rsqcostvalue|rsqmemorylimit|rsqmemoryvalue|rsqwaiters|rsqholders
-------+-------+-------------+-------------+------------+------------+--------------+--------------+----------+----------
(0 rows)
SELECT * FROM gp_toolkit.gp_resq_priority_backend;
rqpsession|rqpcommand|rqppriority|rqpweight
----------+----------+-----------+---------
(0 rows)

-- by default admin_group has concurrency set to -1 which leads to
-- very small memory quota for each resgroup slot, correct it.
ALTER RESOURCE GROUP admin_group SET concurrency 2;
ALTER
