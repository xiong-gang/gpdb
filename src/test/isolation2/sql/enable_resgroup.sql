-- start_ignore
! rmdir /sys/fs/cgroup/cpu/gpdb;
! rmdir /sys/fs/cgroup/cpuacct/gpdb;
! mkdir /sys/fs/cgroup/cpu/gpdb;
! mkdir /sys/fs/cgroup/cpuacct/gpdb;
-- end_ignore

-- enable resource group and restart cluster.
-- start_ignore
! gpconfig -c gp_resource_manager -v group;
! gpconfig -c gp_resource_group_cpu_limit -v 0.5;
! gpstop -rai;
-- end_ignore

SHOW gp_resource_manager;

-- resource queue statistics should not crash
SELECT * FROM pg_resqueue_status;
SELECT * FROM gp_toolkit.gp_resqueue_status;
SELECT * FROM gp_toolkit.gp_resq_priority_backend;

-- by default admin_group has concurrency set to -1 which leads to
-- very small memory quota for each resgroup slot, correct it.
ALTER RESOURCE GROUP admin_group SET concurrency 2;
