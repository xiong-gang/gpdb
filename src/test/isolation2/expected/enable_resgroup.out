-- start_ignore
! rmdir /sys/fs/cgroup/cpu/gpdb;

! rmdir /sys/fs/cgroup/cpuacct/gpdb;

! mkdir /sys/fs/cgroup/cpu/gpdb;

! mkdir /sys/fs/cgroup/cpuacct/gpdb;

! chmod 755 /sys/fs/cgroup/cpu/gpdb;

! chmod 755 /sys/fs/cgroup/cpuacct/gpdb;

-- end_ignore

-- enable resource group and restart cluster.
-- start_ignore
! gpconfig -c gp_resource_manager -v group;
20170502:01:28:12:000367 gpconfig:sdw6:gpadmin-[WARNING]:-Managing queries with resource groups is an experimental feature. A work-in-progress version is enabled.
20170502:01:28:13:000367 gpconfig:sdw6:gpadmin-[INFO]:-completed successfully

! gpstop -rai;
-- end_ignore

SHOW gp_resource_manager;
gp_resource_manager
-------------------
group              
(1 row)
